python/ray/serve/_private/deployment_state.py

```diff
@@ -571,6 +571,19 @@ def check_ready(self) -> Tuple[ReplicaStartupStatus, Optional[str]]:
                     "the replica will be stopped."
                 )
                 return ReplicaStartupStatus.FAILED, str(e.as_instanceof_cause())
+            except RuntimeEnvSetupError as e:
+                msg = (
+                    f"Exception when allocating replica '{self._replica_tag}': {str(e)}"
```

Comment:
Should we use `str(e)` or `traceback.format_exc()` here?

Reply:
As far as I can tell from the examples I ran, they are the same except `traceback.format_exc()` has an extra block that isn't really useful for the user (like what happens when you do `raise from`). Also, `RuntimeEnvSetupError` has its own custom `__str__` method implemented - that's why I decided to go with `str(e)`.

Reply:
Got it, that makes sense. Thanks for the detailed explanation!